tests/cypress/support/commands.js

```diff
@@ -307,6 +314,19 @@ Cypress.Commands.add('waitAllLayersLoaded', ({ queryParams = {}, legacy = false
                 // In legacy drawing with adminId the layer is not added to the layers parameter
                 target += 1
             }
+            // When handling a legacy parameter, the {bod Layer Id} parameter,
+            // which has been reworked into the 'features' layer attribute might add extra
+            // layers, thus the need to check if those extra layers have been added
+            if (legacy && active !== target) {
+                let targetLegacyFeaturePreselection = 0
+                state.layers.activeLayers.forEach((layer) => {
+                    targetLegacyFeaturePreselection += Boolean(
+                        Object.keys(queryParams).includes(layer.id) &&
+                            !queryParams.layers?.includes(layer.geoAdminID)
+                    )
+                })
+                target += targetLegacyFeaturePreselection
+            }
```

Comment:
This is what the check is supposed to do : 
the `legacy && active !== target` is there to exclude non legacy queries (as they can't receive such parameter), and only checks if there is a disparity between the active layers and the targets
It parses trough each layer in "the active layers", and if that layer is inside the query's keys but not in the `layers` param, it adds 1

I will make some tests to check if I can break this, but otherwise it behaved as expected.